### PR TITLE
Migrate presubmits to build01

### PR DESCRIPTION
**Pull request by hongkailiu**
/cc @openshift/openshift-team-developer-productivity-test-platform 

```bash
git --no-pager show ./hack/ensure_job_cluster.py 
commit 18931b2f305e5b1c42e1bd9196810af9a23c0ae3 (HEAD -> mig_presubmits, downstream/mig_presubmits)
Author: Hongkai Liu <hongkailiu@users.noreply.github.com>
Date:   Tue Mar 24 09:51:33 2020 -0400

    Migrate presubmits to build01

diff --git a/hack/ensure_job_cluster.py b/hack/ensure_job_cluster.py
index 73a2e0dff..62832fcaf 100644
--- a/hack/ensure_job_cluster.py
+++ b/hack/ensure_job_cluster.py
@@ -11,6 +11,7 @@ import yaml
 DEFAULT_CLUSTER = "api.ci"
 BUILD01_CLUSTER = "ci/api-build01-ci-devcluster-openshift-com:6443"
 JOB_MAP = {
+    "pull-ci-openshift-ci-tools-master-integration": DEFAULT_CLUSTER,
     "pull-ci-openshift-release-master-build01-dry": DEFAULT_CLUSTER,
     "pull-ci-openshift-release-master-core-dry": DEFAULT_CLUSTER,
     "pull-ci-openshift-release-master-services-dry": DEFAULT_CLUSTER,
@@ -56,6 +57,8 @@ def migrated(file_path):
     # due to https://github.com/openshift/release/pull/7178
     if file_path.endswith('periodics.yaml') and 'openshift/release/' in file_path:
         return False
+    if file_path.endswith('presubmits.yaml'):
+        return True
     return False
 
 def ensure(job_dir, overwrite):
```

/hold

Should we remove the duped jobs (job-name-build01) first?

### CODE REVIEW

**Comment by alvaroaleman** [comment]
> Should we remove the duped jobs (job-name-build01) first?

Yes.

Can we say with confidence that there are no issues, considering all pods got terminated prematurely until yesterday evening by the second sinker?

**Comment by hongkailiu** [comment]
/hold

need to watch for a couple of more days.

**Comment by hongkailiu** [comment]
Will regenerate when https://github.com/openshift/release/pull/7891 merges.

**Comment by hongkailiu** [comment]
Generated list of duplicated jobs with 

```python
#!/usr/bin/env python3
"""This script ensures the cluster of prowjobs is defined as expected.
e.g., python ./hack/ensure_job_cluster.py
"""

import argparse
import logging
import os
import yaml


def get_dup_jobs(file_path, jobs):
    dup_jobs = []
    for job in jobs:
        if any(j["name"] == job["name"] + '-build01' for j in jobs):
            logging.info("found %s", job["name"])
            dup_jobs.append(job["name"])
    return dup_jobs


def ensure(job_dir):
    dup_jobs = []
    for dirpath, _, filenames in os.walk(job_dir):
        for filename in filenames:
            if filename.endswith('.yaml'):
                file_path = os.path.join(dirpath, filename)
                logging.info("handling file %s", file_path)
                with open(file_path) as file:
                    data = yaml.safe_load(file)
                    for job_type in ["presubmits"]:
                        for repo in data.get(job_type, {}):
                            dup_jobs.extend(get_dup_jobs(file_path, data[job_type][repo]))
    logging.info(dup_jobs)


logging.basicConfig(
    level=logging.INFO,
    format="%(asctime)s:%(levelname)s:%(message)s"
    )


def main():
    """main function."""
    logging.info("checking jobs ...")
    parser = argparse.ArgumentParser(description='Ensure prow jobs\' cluster.')
    parser.add_argument('-d', '--job-dir', default='./ci-operator/jobs',
                        help="the path to the job directory")
    args = parser.parse_args()
    ensure(args.job_dir)



if __name__ == "__main__":
    main()
```

```

File: hack/ensure_job_cluster.py [hunk 1]
```diff
@@ -11,21 +11,31 @@
...
+    "pull-ci-openshift-ci-tools-master-integration": DEFAULT_CLUSTER,
     "pull-ci-openshift-release-master-build01-dry": DEFAULT_CLUSTER,
     "pull-ci-openshift-release-master-core-dry": DEFAULT_CLUSTER,
     "pull-ci-openshift-release-master-services-dry": DEFAULT_CLUSTER,
     "periodic-acme-cert-issuer-for-build01": DEFAULT_CLUSTER,
     "periodic-build01-upgrade": BUILD01_CLUSTER,
 }
 
+def load_dup_jobs():
+    """load the duplicated jobs."""
```
Comment (hongkailiu) [inline hack/ensure_job_cluster.py]: if I cannot take the list inline, `pylint` will break.

**Pull request merged**